CMakeLists.txt

```diff
@@ -121,18 +114,77 @@ if(EMSCRIPTEN)
     set(EMSCRIPTEN_FEATURES "${EMSCRIPTEN_FEATURES} -s \"EXTRA_EXPORTED_RUNTIME_METHODS=[ENV']\"")
 endif()
 
-find_package(Clang REQUIRED)
-include(AddLLVM)
-include(HandleLLVMOptions)
-add_definitions(${LLVM_DEFINITIONS})
+find_package(CppInterOp REQUIRED CONFIG PATHS "${CMAKE_PREFIX_PATH}" "${CMAKE_PREFIX_PATH}/lib" "${CPPINTEROP_DIR}" "${CPPINTEROP_DIR}/lib")
+if(CppInterOp_FOUND)
+  set(CppInterOp_CLANG_VERSION "17")
+  message(STATUS "Found CppInterOp: config=${CppInterOp_CONFIG} dir=${CppInterOp_DIR} (found version=${CppInterOp_VERSION} compatible with Clang ${CppInterOp_CLANG_VERSION}.x)")
+endif()
+
 find_package(argparse REQUIRED)
 find_package(pugixml REQUIRED)
 
+set(Python_FIND_VIRTUALENV ONLY)
+find_package(Python COMPONENTS Interpreter Development)
+
+# Configuration
+# =============
+
+if (NOT DEFINED XEUS_CPP_KERNELSPEC_PATH)
+    set(XEUS_CPP_KERNELSPEC_PATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/")
+    set(XEUS_CPP_RESOURCE_DIR ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/clang/${CppInterOp_CLANG_VERSION})
+endif ()
+
+function(configure_kernel kernel)
+  set(XEUS_CPP_CPLUS_INCLUDE_PATH "$ENV{CPLUS_INCLUDE_PATH}")
+  set(XEUS_CPP_PATH "$ENV{PATH}")
+  set(XEUS_CPP_LD_LIBRARY_PATH "$ENV{LD_LIBRARY_PATH}")
+  set(XEUS_CPP_PYTHONPATH "$ENV{PYTHONPATH}")
+  set(XEUS_CPP_VENV_PATH "$ENV{VENV}")
+
+  message(STATUS "Debug: Replace in kernels")
+  message(STATUS "Debug: XEUS_CPP_CPLUS_INCLUDE_PATH=${XEUS_CPP_CPLUS_INCLUDE_PATH}")
+  message(STATUS "Debug: XEUS_CPP_PATH=${XEUS_CPP_PATH}")
+  message(STATUS "Debug: XEUS_CPP_LD_LIBRARY_PATH=${XEUS_CPP_LD_LIBRARY_PATH}")
+  message(STATUS "Debug: XEUS_CPP_PYTHONPATH=${XEUS_CPP_PYTHONPATH}")
+  message(STATUS "Debug: XEUS_CPP_VENV_PATH=${XEUS_CPP_VENV_PATH}")
+  message(STATUS "Debug: CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}")
+  message(STATUS "Debug: CMAKE_CURRENT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}")
+
+  configure_file (
+    "${CMAKE_CURRENT_SOURCE_DIR}/${kernel}/kernel.json.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/${kernel}/kernel.json"
+    )
+  file(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/${kernel}/*.png")
+  foreach(file ${files})
+    configure_file(
+      "${file}"
+      "${CMAKE_CURRENT_BINARY_DIR}/${kernel}/"
+      COPYONLY
+      )
+  endforeach()
+  file(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/${kernel}/*.svg")
+  foreach(file ${files})
+    configure_file(
+      "${file}"
+      "${CMAKE_CURRENT_BINARY_DIR}/${kernel}/"
+      COPYONLY
+      )
+  endforeach()
+endfunction()
+
+file(GLOB _kernels LIST_DIRECTORIES true RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "share/jupyter/kernels/*")
+foreach(_kernel IN LISTS _kernels)
+  message("Configure kernel: ${_kernel}")
+  configure_kernel("${_kernel}")
+endforeach()
```

Comment:
This can go as a separate pull request. We should rename the introduced `CMAKE_` into `XEUS_CPP_` to avoid confusion.